### PR TITLE
Adding JSON-LD/Hydra representations

### LINKED ISSUE #8: Create JSON-LD context documents
I will add static JSON-LD resource(s) for JSON-LD `@context` which will augment the video resources

### PR DESCRIPTION
**Pull request by tpluscode**
I've expanded upon @smizell's work by refactoring the representer code so that common parts are extracted and used but Hydra-rspecific code independently.

As for Hydra, I generate paged collection with 20 items per page and also had to create additional files for JSON-LD context and Hydra API Documentation

fixes #8

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -61,3 +61,10 @@ jekyll-archives:
     month: month-archive
     day: day-archive
     tag: tag-archive
+
+webrick:
+  headers:
+    Link: <http://localhost:4000/hydra/apiDocumentation.jsonld>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"
```
Comment (BigBlueHat) [inline _config.yml]: @tpluscode is there anyway to keep `localhost:4000` out of the config? Or (if it must be there) can we reuse it in the other representations?

File: hydra/@context/video.jsonld [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
+  "@context": {
+    "@vocab": "http://videos.restfest.org/",
+    "@base": "http://purl.org/restfest/videos/",
+    "sch": "http://schema.org/",
+    "xsd": "http://www.w3.org/2001/XMLSchema#",
+    "uri": {
+      "@type": "@id"
+    },
```
Comment (BigBlueHat) [inline hydra/@context/video.jsonld]: This should be `"uri": "@id"`. See https://www.w3.org/TR/json-ld/#h3_aliasing-keywords

File: hydra/@context/video.jsonld [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    "language": {
+      "@id": "sch:inLanguage"
+    },
+    "license": {
+      "@id": "sch:license",
+      "@type": "sch:URL"
+    },
+    "link": {
+      "@type": "@id"
+    }
```
Comment (BigBlueHat) [inline hydra/@context/video.jsonld]: Same situation as with `uri` above. This should be `"link": "@id"`. See https://www.w3.org/TR/json-ld/#h3_aliasing-keywords

File: hydra/video/107304324.jsonld [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
+    "@context": [
+        "http://localhost:4000/hydra/@context/video.jsonld",
+        {
+            "@base": "http://localhost:4000"
```
Comment (BigBlueHat) [inline hydra/video/107304324.jsonld]: Think these `localhost:4000`'s are going to cause problems later.

File: hydra/video/107304324.jsonld [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        }
+    ],
+    "@type": [
+        "http://videos.resftest.org/api/Video",
```
Comment (BigBlueHat) [inline hydra/video/107304324.jsonld]: Typo: s/resftest/restfest/

😄

File: hydra/video/107310631.jsonld [hunk 6]
```diff
@@ -0,0 +1,40 @@
+{
+    "@context": [
+        "http://localhost:4000/hydra/@context/video.jsonld",
+        {
+            "@base": "http://localhost:4000"
+        }
+    ],
+    "@type": [
+        "http://videos.resftest.org/api/Video",
```
Comment (BigBlueHat) [inline hydra/video/107310631.jsonld]: Is this `Video` type defined anywhere? I didn't see it in the [video `@context` file](https://github.com/RESTFest/videos.restfest.org/pull/15/files#diff-bb4b097e9013cbcf878d9d2a7e6eb63d).

File: hydra/@context/video.jsonld (hunk 2 above)
Comment (tpluscode) [reply hydra/@context/video.jsonld]: This is not aliasing, this is https://www.w3.org/TR/json-ld/#type-coercion

File: hydra/@context/video.jsonld (hunk 3 above)
Comment (tpluscode) [reply hydra/@context/video.jsonld]: ditto

File: _config.yml (hunk 1 above)
Comment (tpluscode) [reply _config.yml]: As for we brick's configuration, I don't think it's really relevant. This essentially has effect only when serving locally with `jekyll serve`

File: hydra/video/107310631.jsonld (hunk 6 above)
Comment (tpluscode) [reply hydra/video/107310631.jsonld]: It is defined in the [ApiDocumentation](https://github.com/RESTFest/videos.restfest.org/pull/15/files#diff-225c2b6b462a5df4a60313c6fbd53443R30)

**Pull request merged**